### PR TITLE
fix(specs): provide non clashing names for custom methods

### PR DESCRIPTION
**Pull request by shortcuts**
## 🧭 What and Why

🎟 JIRA Ticket: -

### Changes included:

The current name of the custom methods can clash with reserved keywords of certain languages (e.g. `del` for `python`). This PR proposes a `custom` prefix to those method names.

since Dart is already released, we provide the old methods as `helpers` with a `deprecated` warning. They still leverage the custom methods under the hood.

### CODE REVIEW

**Review by Fluf22 — APPROVED** [review]
Nice 🆗

File: templates/dart/api.mustache [hunk 1]
```diff
@@ -157,7 +157,45 @@ final class {{classname}} implements ApiClient {
   }  
   {{/operation}}
 
+  {{^isAlgoliasearchClient}}
```
Comment (millotp) [inline templates/dart/api.mustache:160]: I think this should be for all clients, not just != algoliasearch

**Review by millotp — COMMENTED** [review]
looks great ! small comment about dart, I though we included the custom requests for algoliasearch too

File: templates/dart/api.mustache (hunk 1 above)
Comment (shortcuts) [reply templates/dart/api.mustache:160]: I don't think so for dart but TBC!

Comment (shortcuts) [reply templates/dart/api.mustache:160]: confirmed they are not in the algoliasearch client

Comment (millotp) [reply templates/dart/api.mustache:160]: okay but why is it removed and not replaced in the searchClient ? https://github.com/algolia/api-clients-automation/commit/d0a0c96d195bb632c61861f46bbaa01eeda04503#diff-6e097578fa886b0b109622bbfadeda83cc72b0c60fba70d7d8a6c429b64358faL55

Comment (shortcuts) [reply templates/dart/api.mustache:169]: done here https://github.com/algolia/api-clients-automation/pull/2369/commits/7c3fbf6495dc479a10fac35508760dced9742a3d

**Pull request merged**